crates/store/src/db/models/queries/transactions.rs

```diff
@@ -126,14 +173,191 @@ pub struct TransactionSummaryRowInsert {
+///
+/// # Raw SQL
+/// ```sql
+/// SELECT
+///     account_id,
+///     block_num,
+///     transaction_id,
+///     initial_state_commitment,
+///     final_state_commitment,
+///     input_notes,
+///     output_notes,
+///     size_in_bytes
+/// FROM transactions
+/// WHERE block_num > ?1 AND block_num <= ?2 AND account_id IN rarray(?3)
+/// ORDER BY block_num ASC
+/// LIMIT ?4 OFFSET ?5
+/// ```
+/// Note: The query is executed in chunks of 1000 transactions to prevent loading
+/// excessive data and to stop as soon as the accumulated size approaches the 5MB limit.
+pub fn select_transactions_records(
+    conn: &mut SqliteConnection,
+    account_ids: &[AccountId],
+    block_range: RangeInclusive<BlockNumber>,
+) -> Result<(BlockNumber, Vec<crate::db::TransactionRecord>), DatabaseError> {
+    const MAX_PAYLOAD_BYTES: usize = 5 * 1024 * 1024; // 5 MB
+    const CHUNK_SIZE: i64 = 1000; // Read 1000 transactions at a time
+
+    QueryParamAccountIdLimit::check(account_ids.len())?;
+
+    if block_range.is_empty() {
+        return Err(DatabaseError::InvalidBlockRange {
+            from: *block_range.start(),
+            to: *block_range.end(),
+        });
+    }
+
+    let desired_account_ids = serialize_vec(account_ids);
+
+    // Read transactions in chunks to prevent loading excessive data and to stop
+    // as soon as we approach the size limit
+    let mut all_transactions = Vec::new();
+    let mut current_size = 0i64;
+    let mut offset = 0i64;
+
+    loop {
+        let chunk =
+            SelectDsl::select(schema::transactions::table, TransactionRecordRaw::as_select())
+                .filter(schema::transactions::block_num.gt(block_range.start().to_raw_sql()))
+                .filter(schema::transactions::block_num.le(block_range.end().to_raw_sql()))
+                .filter(schema::transactions::account_id.eq_any(&desired_account_ids))
+                .order((schema::transactions::block_num.asc(),))
+                .offset(offset)
+                .limit(CHUNK_SIZE)
+                .load::<TransactionRecordRaw>(conn)
+                .map_err(DatabaseError::from)?;
+
+        // If no more data, we're done
+        if chunk.is_empty() {
+            break;
+        }
+
+        // Calculate the size of this chunk
+        let chunk_size: i64 = chunk.iter().map(|tx| tx.size_in_bytes).sum();
+
+        // If adding this chunk would exceed the limit, we need to be more careful
+        if current_size + chunk_size > MAX_PAYLOAD_BYTES as i64 {
+            // Add transactions from this chunk one by one until we hit the limit
+            for tx in chunk {
+                if current_size + tx.size_in_bytes <= MAX_PAYLOAD_BYTES as i64 {
+                    current_size += tx.size_in_bytes;
+                    all_transactions.push(tx);
+                } else {
+                    // Can't fit this transaction, stop here
+                    break;
+                }
+            }
+            break;
+        }
```

Comment:
Maybe for simplicity we always add transactions one-by-one. It will be a bit less efficient, but I think the difference should be pretty small. Basically, the code could be something like:
```rust
// Add transactions from this chunk one by one until we hit the limit
for tx in chunk {
    if current_size + tx.size_in_bytes <= MAX_PAYLOAD_BYTES as i64 {
        current_size += tx.size_in_bytes;
        all_transactions.push(tx);
    } else {
        // Can't fit this transaction, stop here
        break;
    }
}
```
And then we don't need the code on lines 321 - 330 below.